parboiled-core/src/main/scala-3/org/parboiled2/ParserMacros.scala

```diff
@@ -868,35 +855,41 @@ base match {
           case _                 => reportError(s"Illegal `separatedBy` base: $base", base)
         }
 
-      case x =>
-        x.asTerm.underlyingArgument match {
-          // cannot easily be converted because we would have to list all ActionOps instances
-          case Apply(
-                Apply(
-                  TypeApply(
-                    Select(
-                      Select(Apply(Apply(TypeApply(Select(_, "rule2ActionOperator"), _), List(base)), _), "~>"),
-                      "apply"
-                    ),
-                    _
-                  ),
-                  List(body)
+      case '{ ($p: Parser).run[t]($e)($l) } => RunAction(e)
+    }
+
+    lazy val rules1PF: PartialFunction[Term, OpTree] = {
```

Comment:
What is the meaning of the `0` and the `1`? Is it about the priority?

Reply:
Yeah, basically it's about priority. First the `Expr` is matched and if nothing is found a `Term` is matched afterwards. it's mostly about code organization.

I had to reorganize this bit because in `expand` below, I need to be able to check whether a given expression can be compiled as a rule or whether it should be taken as is (i.e. a complex expression returning a rule).